### DISCUSSION TITLE
New dark theme

### DISCUSSION THREAD

**Discussion by simurai** [primer/github-vscode-theme#114] [General]
In version `2.0.0` the `GitHub Dark` theme changed the colors to match the `Dark` theme used on github.com.

![github-theme](https://user-images.githubusercontent.com/378023/105256931-0f1cd000-5bca-11eb-8ac1-725ce933cf60.png)

We got some feedback that the colors are [too dark/contrasting](https://github.com/primer/github-vscode-theme/issues/108) or the [old version should be kept](https://github.com/primer/github-vscode-theme/issues/112). We understand that a bigger change like that can feel disruptive. Here quick FAQ:

## Why change the dark colors?

The main goal of this "official" GitHub theme is to be very un-opinionated and keep it as close to github.com as possible. So it should follow the themes available on github.com's [settings](https://github.com/settings/appearance). This keeps things simple and should avoid confusion.

## Why not keep it with a different name?

We were considering that, but then users expect it to be an actively maintained theme just like the `Light` and new `Dark`. Again keeping this "official" GitHub theme closely aligned with what users see under github.com's [settings](https://github.com/settings/appearance) seemed the right way forward.

## Can I still use the "old" dark theme?

Yes, you can still install it via `.vsix` from the [github-vscode-theme-dark-classic](https://github.com/primer/github-vscode-theme-dark-classic) repo. See [Install](https://github.com/primer/github-vscode-theme-dark-classic#install) in the README. Note that we don't plan to update it any further though. But if someone wants to fork and publish it on the marketplace, that's ok too.

**Update**: It's now also available from the [marketplace](https://marketplace.visualstudio.com/items?itemName=BerriJ.github-vscode-theme-dark-classic).

## Will there be more themes?

Yes, the plan is to add more themes. We're already testing a "Dimmed" version in this PR: https://github.com/primer/github-vscode-theme/pull/105

**Comment by BerriJ** [comment]
I just forked and published it. It's available [here](https://marketplace.visualstudio.com/items?itemName=BerriJ.github-vscode-theme-dark-classic). @primer Thank you so much for the permission to do so.

Note that I have nearly no experience programming VSC Extensions. So I'm happy to invite collaborators and merge PR's as Issues come up.

**Reply by simurai** [reply]
Just tried to install it from the marketplace and everything worked fine. ✌️

**Comment by chadluo** [comment]
If we want colors to match between the website and the editor, can we use the editor theme (old) on the website?

I can understand that the team want to unify the color themes across the website and the editor, but I didn't like the dark website. I tried that for a short while, found it too dark and too much contrast for me, and just switched back without leaving any comments/feedback.

From people's comments it seems that the old editor theme is really popular. I don't know the statistics or feedback but I think it's probably feasible backporting the color theme.

**Reply by simurai** [reply]
That could be an option. We plan is to add more themes, but we're still exploring which ones to add.

**Comment by simurai** [comment]
Since releasing the `2.0.0` version we got a lot of feedback asking to keep the old dark theme. So we decided to rollback all the changes to the Dark theme (also some to Light) and publish it as `3.0.0`. This should be the same as `1.1.5`.

A special thanks ❤️ to @BerriJ for publishing the `GitHub Dark Classic` theme to ease our misstep. Feel free to keep it published if you like.

#### What about the dark theme from github.com?

We're still evaluating the best path forward. An option could be to release [all the themes from github.com](https://github.com/settings/appearance) under a different name. For example `GitHub.com Dark`.

**Reply by manuelricci** [reply]
I love the new theme, I hope you'll publish it separately or give us the VSX installation

**Comment by krailler** [comment]
I'll keep on 2.0.1, I love dark mode in this version

**Comment by kainbacher** [comment]
I love the gray one. On the black theme the contrast is too much. so +1 for 3.0.0

**Comment by arturogh** [comment]
Would love a preview for each of the 5 options :). Amazing themes btw, thanks for putting these out there.

**Reply by dominicegginton** [reply]
Do you mean the preview image in the README? I think GitHub have dropped support in favour of the new themes implemented in #132, that's probably why the old themes are not displayed.

**Reply by simurai** [reply]
Maybe not _entirely_ dropped. 😄  The `GitHub Light` + `GitHub Dark` themes are still available as part of this extension, but it doesn't really fit in with the rest of [github.com's themes](https://github.com/settings/appearance):

![image](https://user-images.githubusercontent.com/378023/115527756-50f20b00-a2cc-11eb-9324-047e064501f3.png)

So they are not really showcased anymore and more seen as "legacy" themes.

**Reply by arturogh** [reply]
I mean, having previews or renderings of each theme. Sure, you can choose each theme, one by one, to see what each looks like, but having the previews somewhere would be cool, specially as the differences are subtle (yet important :) ).
